stock_orderpoint_generator/models/orderpoint_template.py

```diff
@@ -82,6 +82,20 @@ class OrderpointTemplate(models.Model):
     qty_forecast = fields.Float(compute=False, store=False)
     product_category_id = fields.Many2one(related=False, store=False)
 
+    allowed_warehouse_ids = fields.One2many(
+        comodel_name="stock.warehouse", compute="_compute_allowed_warehouse_ids"
+    )
+
+    @api.depends("company_id")
+    def _compute_allowed_warehouse_ids(self):
+        for template in self:
+            domain = [
+                "|",
+                ("company_id", "=", False),
+                ("company_id", "=", template.company_id.id),
+            ]
+            template.allowed_warehouse_ids = self.env["stock.warehouse"].search(domain)
```

Comment:
Better to do the search outside the loop and then grouping templates per company and filter.